common/core/hwclayer.cpp

```diff
@@ -178,14 +178,18 @@ void HwcLayer::SetReleaseFence(int32_t fd) {
 }
 
 int32_t HwcLayer::GetReleaseFence() {
-  if (!sf_handle_)
-    return -1;
   int32_t old_fd = release_fd_;
   release_fd_ = -1;
   return old_fd;
 }
 
 void HwcLayer::SetAcquireFence(int32_t fd) {
+  if (!sf_handle_) {
+    if (fd > 0) {
+      close(fd);
+    }
+    acquire_fence_ = -1;
+  }
   if (acquire_fence_ > 0) {
```

Comment:
In GetAcquireFence(), if sf_handle_ is invalid, does it need to close fd before exiting?

int32_t HwcLayer::GetAcquireFence() {
  if (!sf_handle_) {
    if (fd > 0) {
      close(fd);
    }
    return -1;
}

Reply:
no need